File: proposed-projects/round-trip-experiment/reverse/20260909T134433Z_3f62598c/20260910T080424Z_8686d09e/ported_implementation/src/rulesBuilder/parseSpace.ts
import { charAt } from '../utils/charAt.js';

export const parseSpace = (src: string, pos: number, newlineOk: boolean): number => {
  const isSpace = (char: string): boolean => char === ' ' || char === '\t' || char === '#' ||
    (newlineOk && (char === '\r' || char === '\n'));

  while (isSpace(charAt(src, pos))) {
    if (charAt(src, pos) === '#') {
      while (charAt(src, pos) && charAt(src, pos) !== '\r' && charAt(src, pos) !== '\n') {
        pos += 1;
      }
    } else {
      pos += 1;
    }
  }
  return pos;
};

File: packages/gbnf-experiment/data/20260909T015142Z_f8ca89ff/ported_implementation/rules_builder/parse_space.py
from __future__ import annotations

from ..utils.js_compat import char_at

__all__ = ["parse_space", "parseSpace"]


def parse_space(src: str, pos: int, newline_ok: bool) -> int:
    while char_at(src, pos) in (" ", "\t", "#") or (
        newline_ok and char_at(src, pos) in ("\r", "\n")
    ):
        if char_at(src, pos) == "#":
            while char_at(src, pos) and char_at(src, pos) not in ("\r", "\n"):
                pos += 1
        else:
            pos += 1
    return pos


parseSpace = parse_space
